framework/Web/UI/TClientScriptManager.php

```diff
@@ -576,11 +576,12 @@ public function getStyleSheetCodes()
 	 * Registers a javascript file in the page head
 	 * @param string $key a unique key identifying the file
 	 * @param string $url URL to the javascript file
+	 * @param bool $async load the javascript file asynchronously, default false
 	 */
-	public function registerHeadScriptFile($key, $url)
+	public function registerHeadScriptFile($key, $url, $async = false)
 	{
 		$this->checkIfNotInRender();
-		$this->_headScriptFiles[$key] = $url;
+		$this->_headScriptFiles[$key] = $async ? [$url, $async] : $url;
```

Comment:
This probably breaks rendering of new script files during an ajax call, currently used by at least THtmlArea.
On callback, TActivePageAdapter::renderResponse() retrieves a list of the javascript files and renders their names inside a <X-PRADO-SCRIPTLIST> tag that gets checked by clientside javascript for missing scripts.
The clientside javascript expect plain urls.